test/val/val_memory_test.cpp

```diff
@@ -2119,9 +2119,9 @@ OpFunctionEnd
   EXPECT_EQ(SPV_ERROR_INVALID_ID, ValidateInstructions(SPV_ENV_WEBGPU_0));
   EXPECT_THAT(
       getDiagnosticString(),
-      HasSubstr("For WebGPU, OpTypeStruct variables containing "
-                "OpTypeRuntimeArray must have storage class of StorageBuffer\n "
-                " %6 = OpVariable %_ptr_Uniform__struct_3 Uniform\n"));
+      HasSubstr(
```

Comment:
This seems like the wrong update to make. I'd suggest instead changing `%struct_t` to be a Block in StorageBuffer instead.